src/trigger/HSISourceModel.hpp

```diff
@@ -0,0 +1,122 @@
+ * received with this code.
+ */
+#ifndef TRIGGER_PLUGINS_HSISOURCEMODEL_HPP_
+#define TRIGGER_PLUGINS_HSISOURCEMODEL_HPP_
+
+#include <functional>
+#include "readoutlibs/concepts/SourceConcept.hpp"
+#include "detdataformats/DetID.hpp"
+#include "dfmessages/HSIEvent.hpp"
+#include "triggeralgs/TriggerCandidate.hpp"
+#include "trigger/TCWrapper.hpp"
+
+
+#include "iomanager/IOManager.hpp"
+#include "iomanager/Sender.hpp"
+#include "iomanager/Receiver.hpp"
+#include "logging/Logging.hpp"
+#include "coredal/DaqModule.hpp"
+#include "appdal/DataSubscriber.hpp"
+#include "appdal/HSI2TCTranslatorConf.hpp" 
+#include "appdal/HSISignalWindow.hpp" 
+
+namespace dunedaq::trigger {
+
+
+class HSISourceModel : public readoutlibs::SourceConcept
+{
+public: 
+  using inherited = readoutlibs::SourceConcept;
+
+  /**
+   * @brief SourceModel Constructor
+   * @param name Instance name for this SourceModel instance
+   */
+  
+  HSISourceModel(): readoutlibs::SourceConcept() {}
+  ~HSISourceModel() {}
+
+  void init(const coredal::DaqModule* cfg) override {
+    if (cfg->get_outputs().size() != 1) {
+      throw readoutlibs::InitializationError(ERS_HERE, "Only 1 output supported for subscribers");
+    }
+    m_data_sender = get_iom_sender<trigger::TCWrapper>(cfg->get_outputs()[0]->UID());
+
+    if (cfg->get_inputs().size() != 1) {
+      throw readoutlibs::InitializationError(ERS_HERE, "Only 1 input supported for subscribers");
+    }
+    m_data_receiver = get_iom_receiver<dfmessages::HSIEvent>(cfg->get_inputs()[0]->UID());
+    auto data_reader = cfg->cast<appdal::DataSubscriber>();
+    if (data_reader == nullptr) {
+       throw readoutlibs::InitializationError(ERS_HERE, "DAQ module is not a DataReader");
+    }
+    auto hsi_conf = data_reader->get_configuration()->cast<appdal::HSI2TCTranslatorConf>();
+    if (hsi_conf == nullptr) {
+	throw readoutlibs::InitializationError(ERS_HERE, "Missing HSI2TCTranslatorConf");
+    }
+    for (auto win : hsi_conf->get_signals()) {
+	   m_signals[win->get_signal_type()] = std::pair<uint32_t, uint32_t>(win->get_time_before(), win->get_time_after());
+    }
+  }
+
+  void start() {
+    m_data_receiver->add_callback(std::bind(&HSISourceModel::handle_payload, this, std::placeholders::_1));
+  }  
+
+  void stop() {
+    m_data_receiver->remove_callback();
+  }
+
+  void get_info(opmonlib::InfoCollector& /*ci*/, int /*level*/) {}
+
+  bool handle_payload(dfmessages::HSIEvent& data) // NOLINT(build/unsigned)
+  {
+    
+    triggeralgs::TriggerCandidate candidate;
+    auto signal_info = m_signals.find(data.signal_map);
+    if (signal_info != m_signals.end()) {
+        // clang-format off
+        candidate.time_start = data.timestamp - signal_info->second.first;  // time_start
+        candidate.time_end   = data.timestamp + signal_info->second.second; // time_end,
+        // clang-format on    
+    } else {
+        throw dunedaq::trigger::SignalTypeError(ERS_HERE, "HSI subscriber" , data.signal_map);
+    }
```

Comment:
`hsi_passthrough` option is missing here, which passes the `data` (`HSIEvent`) with any `signal_map` through the first `m_signals` without checking if the signal map is inside our local `m_signals`.

Reply:
we can add back this logics (when merging feature/OKS into develop), if really needed, but I don't see the added value of it